stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/Message.kt

```diff
@@ -30,18 +30,25 @@ internal fun Message.updateUsers(users: Map<String, User>): Message =
 
 /**
  * Fills [Message.mentionedUsersIds] based on [Message.text] and [Channel.members].
+ *
+ * It combines the users found in the input with pre-set [Message.mentionedUsersIds], in case someone
+ * is manually added as a mention. Currently only searches through the channel members for possible mentions.
+ *
+ * @param channel The channel whose members we can check for the mention.
  */
 internal fun Message.populateMentions(channel: Channel) {
     if ('@' !in text) {
         return
     }
 
     val text = text.lowercase()
-    mentionedUsersIds = channel.members.mapNotNullTo(mutableListOf()) { member ->
+    val mentions = mentionedUsersIds.toMutableSet() + channel.members.mapNotNullTo(mutableListOf()) { member ->
```

Comment:
This change could also have the benefit of allowing library users to call `ChatDomain.sendMessage()/editMessage()` manually from a custom `MessageSendHandler`, populating any `mentionedUserIds` they'd like.

Reply:
Yeah, this is the main update to allow mentions to be populated externally + with the input, so it allows for external `mentionedUserIds` to be set.

But it has the added benefit that you described!